final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,97 @@
+import argparse
+import xml.etree.ElementTree as ET
+import requests
+import json
+import logging
+
+import rss_reader.ClassNews as ClassNews
+
+VERSION = 1.1
+
+
+def my_parser():
+    # Parse our arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument("sourse", help="RSS URL")
+    parser.add_argument('--version', action='store_true', help='Print version info')
+    parser.add_argument('--json', action='store_true', help='Print result as JSON in stdout')
+    parser.add_argument('--verbose', action='store_true', help='Outputs verbose status messages')
+    parser.add_argument('--limit', type=int, help='Limit news topics if this parameter provided')
+
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    try:
+        args = my_parser()
+        logging_level = logging.CRITICAL
+        if args.verbose:
+            logging_level = logging.INFO
+        if args.version:
+            print("Current version: " + str(VERSION))
+        if args.limit:
+            print('News LIMIT: ' + str(args.limit))
+
+        logging.basicConfig(format='%(levelname)s:%(message)s', level=logging_level)
+
+        # Get request
+        logging.info('Start parsing')
+        rss_request = requests.get(args.sourse)
+
+        # to check ReadTimeout exception
+        #rss_request = requests.get(args.sourse, timeout=(1, 0.01))
+
+        # Check status code
+        status_code = rss_request.status_code
+        logging.info("Status code {}".format(status_code))
+        # if status_code == 404:
+        #     raise requests.exceptions.HTTPError
+        rss_request.raise_for_status()
+
+        logging.info('Parsing completed successfully')
+
+        # Here we check the type of response. To correctly process it
+        if rss_request.headers['content-type'] == "application/xml":
+            root = ET.fromstring(rss_request.content)
+
+            # Here we get title of api
+            for channel_info in root.iter('channel'):
+                for item in channel_info:
+                    if item.tag == 'title':
+                        main_title = item.text
+
+            # Here we have the dictionary of articles
+            res_dict_articles = ClassNews.xml_arguments_for_class(root, args.limit)
+            #print(type(res_dict_articles))
+
+            logging.info('Print news:')
+            print("\nFeed: {}".format(main_title))
```

Comment:
There is a case existing, when `main_title` may not be initialized to this moment.